manifests/pool6.pp

```diff
@@ -1,24 +1,20 @@
 # == Define: dhcp::pool6
 #
 define dhcp::pool6 (
-  $network,
-  $prefix,
-  $range            = '',
-  $range_temp       = '',
-  $failover         = '',
-  $options          = '',
-  $parameters       = '',
-  $nameservers      = undef,
-  $nameservers_ipv6 = undef,
-  $pxeserver        = undef,
-  $mtu              = undef,
-  $domain_name      = '',
-  $ignore_unknown   = undef,
+  String $network,
+  String $prefix,
+  $range                  = '',
```

Comment:
nope. so prefix is just the prefixlength, so int is fine. the ipv6 regex in stdlib only matches for complete host addresses, not prefixes (for example 2001:db8::). Also the regex looks way to short.